monitor/xfeemngr/tokenprice/buffer.go

```diff
@@ -44,16 +55,23 @@ func (b *Buffer) Price(token tokens.Token) float64 {
 
 // Stream starts streaming prices for all tokens into the buffer.
 func (b *Buffer) Stream(ctx context.Context) {
+	if b.streaming {
+		return
+	}
```

Comment:
This allows for multiple entries to `Stream`. If you invoke it fast enough you can have multiple instances of the `if` expression being evaluated to false because they haven't reached the end of the func yet. Therefore, this `if` is probably not "safe" enough.

Reply:
If you want this to be called only once, then better use `sync.Once`